### PR TITLE
Integration test scenario for Service Bus binder

### PR DESCRIPTION
**Pull request by ZhuXiaoBing-cn**
* Add ServiceBus binder integration tests.

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -178,7 +178,7 @@ com.azure.spring:azure-spring-boot-test-cosmosdb;1.0.0;1.0.0
 com.azure.spring:azure-spring-boot-test-keyvault;1.0.0;1.0.0
 com.azure.spring:azure-spring-boot-test-keyvault-reactive;1.0.0;1.0.0
 com.azure.spring:azure-spring-boot-test-parent;1.0.0;1.0.0
-com.azure.spring:azure-spring-boot-test-servicebus-jms;1.0.0;1.0.0
+com.azure.spring:azure-spring-cloud-test-servicebus-binder;1.0.0;1.0.0
```
Comment (saragluna) [inline eng/versioning/version_client.txt:181]: the azure-spring-boot-test-servicebus-jms is still needed to test our JMS starter.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                LOGGER.info("New message received: '{}'", message);
+                Assertions.assertEquals(message.getPayload(), ServiceBusQueueAndTopicBinderIT.message);
+                count.addAndGet(1);
+            };
+        }
+    }
+
+    @Test
+    public void testSendAndReceiveMessage() throws InterruptedException {
+        Thread.sleep(1500);
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java:93]: is there a way to get rid of this `sleep` from our test case?
does other spring cloud stream binder use same way to do testing, for example the community version kafka stream?

Comment (ZhuXiaoBing-cn) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java]: Refer to Kafka Stream and use `CountDownlatch` to get rid of `sleep`.

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/pom.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.junit.vintage</groupId>
+          <artifactId>junit-vintage-engine</artifactId>
+        </exclusion>
```
[the remark this answers is not in the document]
Comment (ZhuXiaoBing-cn) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/pom.xml]: Yes, I have removed it.

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java [hunk 4]
```diff
@@ -34,7 +33,7 @@
 
     private static String message = UUID.randomUUID().toString();
 
-    private static final AtomicInteger count = new AtomicInteger(0);
+    public static CountDownLatch latch = new CountDownLatch(2);
```
Comment (saragluna) [inline sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/java/com/azure/spring/sample/servicebus/binder/ServiceBusQueueAndTopicBinderIT.java:36]: This should not be `public`

**Review by saragluna — CHANGES_REQUESTED** [review]

**Review by saragluna — APPROVED** [review]
LGTM. @ZhuXiaoBing-cn, let's create another PR to use different connection strings for two binnders.

**Comment by ZhuXiaoBing-cn** [comment]
#21966

**Pull request merged**